### PR TITLE
Devtooling 192

### PR DESCRIPTION
**Pull request by kavinbalagen**

### CODE REVIEW

File: genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	registrar "terraform-provider-genesyscloud/genesyscloud/resource_register"
+)
+
+func SetRegistrar(l registrar.Registrar) {
+	l.RegisterDataSource("genesyscloud_telephony_providers_edges_edge_group", DataSourceEdgeGroup())
+
+	l.RegisterResource("genesyscloud_telephony_providers_edges_edge_group", ResourceEdgeGroup())
+
+	l.RegisterExporter("genesyscloud_telephony_providers_edges_edge_group", EdgeGroupExporter())
+
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init.go:13]: you dont need this whitespace

Comment (dginty4) [reply genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init.go:13]: Actually scratch that, what is this file? `SetRegistrar` is generally in the schema file

File: genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+func (r *registerTestInstance) registerTestResources() {
+
+	r.resourceMapMutex.Lock()
+	defer r.resourceMapMutex.Unlock()
+
+	providerResources["genesyscloud_telephony_providers_edges_edge_group"] = ResourceEdgeGroup()
+
+	// external package dependencies for outbound
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go:29]: Comment need to be updated

File: genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+}
+
+func (r *registerTestInstance) registerTestDataSources() {
+
+	r.datasourceMapMutex.Lock()
+	defer r.datasourceMapMutex.Unlock()
+
+	providerDataSources["genesyscloud_telephony_providers_edges_edge_group"] = DataSourceEdgeGroup()
+	// external package dependencies for outbound
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go:43]: same here

File: genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	providerDataSources["genesyscloud_telephony_providers_edges_site"] = edgeSite.DataSourceSite()
+	providerDataSources["genesyscloud_telephony_providers_edges_trunkbasesettings"] = telephony.DataSourceTrunkBaseSettings()
+
+}
+
+func initTestresources() {
+	providerDataSources = make(map[string]*schema.Resource)
+	providerResources = make(map[string]*schema.Resource)
+
+	reg_instance := &registerTestInstance{}
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go:53]: Best to use camelcase here

File: genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+	reg_instance.registerTestDataSources()
+	reg_instance.registerTestResources()
+
+}
+
+func TestMain(m *testing.M) {
+
+	// Run setup function before starting the test suite for Outbound Package
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_init_test.go:62]: Same here, comment needs to be updated

**Review by charliecon — COMMENTED** [review]
I would move the schema definitions for the edge group exporter, data source, and resource into a separate file called `resource_genesyscloud_telephony_providers_edges_edge_group_schema.go`. Also, as Declan mentioned, we keep the SetRegistrar func in this _schema file. This `init.go` file only really exists in packages that don't follow the format we're going for, like the telephony package that just has multiple resources dumped into it 

All we want to have in the file `resource_genesyscloud_telephony_providers_edges_edge_group.go` is the functions for CRUD operations, and the getAll function used by the exporter.. so you could also move the other helper functions in that file into a file called `resource_genesyscloud_telephony_providers_edges_edge_group_utils.go`

File: genesyscloud/telephony_providers_edges_edge_group/resource_genesyscloud_telephony_providers_edges_edge_group.go [hunk 6]
```diff
@@ -215,7 +163,7 @@ func readEdgeGroup(ctx context.Context, d *schema.ResourceData, meta interface{}
 		}
 		d.Set("phone_trunk_base_ids", nil)
 		if edgeGroup.PhoneTrunkBases != nil {
-			d.Set("phone_trunk_base_ids", flattenPhoneTrunkBases(*edgeGroup.PhoneTrunkBases))
+			d.Set("phone_trunk_base_ids", FlattenPhoneTrunkBases(*edgeGroup.PhoneTrunkBases))
```
Comment (HemanthDogiparthi12) [inline genesyscloud/telephony_providers_edges_edge_group/resource_genesyscloud_telephony_providers_edges_edge_group.go:166]: is this used outside package. ?

**Review by HemanthDogiparthi12 — APPROVED** [review]

File: genesyscloud/telephony_providers_edges_edge_group/resource_genesyscloud_telephony_providers_edges_edge_group.go (hunk 6 above)
Comment (kavinbalagen) [reply genesyscloud/telephony_providers_edges_edge_group/resource_genesyscloud_telephony_providers_edges_edge_group.go]: It is not, i have changed it.

**Review by dginty4 — APPROVED** [review]

**Pull request merged**